### ISSUE TITLE
Disable var-naming rule if type is implementing generated interface

### ISSUE THREAD

**Issue by menma1234** [mgechev/revive#748]
**Is your feature request related to a problem? Please describe.**

We use a library that generates some interfaces that need to be implemented ([gqlgen](https://github.com/99designs/gqlgen/)) and it will generate interfaces that don't necessarily conform to the `var-naming` rule\, e.g. `Vm`.

When creating types that implement these interfaces, the function implementation name gets flagged by the `var-naming` rule.

**Describe the solution you'd like**

It'd be nice if `revive` could automatically detect when a type implements an interface that's defined within a generated file and ignore this. We don't really have much of a choice in the naming since the interface itself is generated by an external library.

**Describe alternatives you've considered**

We could add `Vm` to the list of allowed names within the rule configuration, but we still would like to have the code flagged when the naming doesn't conform elsewhere in our code.

Similarly, we could add the `disable` comment to those implementations but that just looks ugly.

**Additional context**

As a test:

In `main.go`:

```go
package main

import "fmt"

type SomeStruct struct{}

func (s *SomeStruct) VmInfo() string {
        return "asdf"
}

func run(i SomeInterface) {
        fmt.Println(i.VmInfo())
}

func main() {
        run(&SomeStruct{})
}
```

In the generated interface:

```go
// Code generated by some lib, DO NOT EDIT.

package main

type SomeInterface interface {
        VmInfo() string
}
```

Running `revive` results in:

```
$ revive -config config.toml -formatter friendly .
  ⚠  https://revive.run/r#var-naming  method VmInfo should be VMInfo
  main.go:7:22

⚠ 1 problem (0 errors, 1 warning)

Warnings:
  1  var-naming
```

Config:

```toml
[rule.var-naming]
```

**Comment by chavacava** [comment]
Hi @menma1234, thanks for filling the issue.

Unfortunately, solving the problem you describe requires a global analysis of the project's code base and `revive`, for the sake of performance and simplicity,  does only file-scoped analysis.

Further, I think the naming problem of the generated code should be addressed at the generation library: make the gen lib generate GO idiomatic identifiers.
